### PR TITLE
Types, round 2

### PR DESCRIPTION
**Pull request by pudo**
This PR aims to type-annotate the key FtM schema object, like `Model`, `Schema`, `Property` and `EntityProxy` :) 

fc90a70c6abd68dd5d0954bcf7cff7df

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 test:
 	nosetests --with-coverage --cover-package=followthemoney --cover-erase --cover-html --cover-html-dir=coverage-report
 
+types:
+	mypy --strict followthemoney/types \
+		followthemoney/schema.py \
+		followthemoney/property.py \
+		followthemoney/proxy.py \
+		followthemoney/graph.py \
+		followthemoney/model.py
```
Comment (Rosencrantz) [inline Makefile:16]: Is the plan here that we add more and more files to this list as they're converted to be typed?

File: followthemoney/graph.py [hunk 2]
```diff
@@ -52,22 +66,22 @@ def to_dict(self):
...
+    def __repr__(self) -> str:
         return "<Node(%r, %r, %r)>" % (self.id, self.type, self.caption)
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash(self.id)
 
-    def __eq__(self, other):
-        return self.id == other.id
+    def __eq__(self, other: Any) -> bool:
```
Comment (Rosencrantz) [inline followthemoney/graph.py:83]: Any?

File: followthemoney/property.py [hunk 3]
```diff
@@ -1,21 +1,71 @@
...
+
+
+class PropertyToDict(PropertyDict, total=False):
+    name: str
+    qname: str
+    reverse: Optional[str]
+    stub: Optional[bool]
+
+
+class Property:
```
Comment (Rosencrantz) [inline followthemoney/property.py:41]: Might be worth defining a standard here as I've been putting the root type class at the top. Not bothered where it lives, but I do like consistency.

File: followthemoney/graph.py (hunk 2 above)
Comment (pudo) [reply followthemoney/graph.py:83]: That's a suggestion by mypy, pretty weird.

File: Makefile (hunk 1 above)
Comment (pudo) [reply Makefile:16]: Yep. It establishes an explicit coverage as long as the whole package isn't compliant.

**Review by Rosencrantz — APPROVED** [review]
This thing is massive and I'm not, in all honestly going to be able to give it an appropriate review - testament to the fact that changes should be small and incremental I guess - with the time I have. 

I'm going to make the assumption that, with tests passing etc that this change will work as expected. I'll likely live to regret this decision

**Comment by pudo** [comment]
I can make a commitment to be around to help sort out the fallout visible in the open. For the datavault, let's just pray.

**Comment by pudo** [comment]
Merged via #479.

**Pull request closed without merging**